README.md

```diff
@@ -1,9 +1,310 @@
+
+The following can be passed to `osde2e` executable as environment variables.
+
+
+* `ADDON_TEST_USER`: The in-cluster user that the test harness containers will run as. Allows for a single wildcard (`%s`) which will automatically be evaluated as the namespace for the test harness.
+* `ADDON_IDS`: Comma-delimited list of addons to install. e.g.
+  ```yaml
+      - name: ADDON_IDS
+        value: managed-api-service
+  ```
+* `ADDON_TEST_HARNESSES`: Comma delimited list of docker images to run within the test namespace.
+* `ADDON_PARAMETERS` allows you to pass parameters to your addon. The format is a two-level JSON object. The outer object's keys are the IDs of addons, and the inner objects are key-value pairs that will be passed to the associated addon. e.g.
+  ```yaml
+    - name: ADDON_PARAMETERS
+      value: '{
+       "managed-api-service":{
+         "addon-managed-api-service": "1",
+         "addon-resource-required":"true", 
+         "cidr-range": "10.1.0.0/26"}
+       }'
+  ```
+* ROSA variables
+  * `ROSA_AWS_REGION`: (we recommend setting this to `random`)
+  * `ROSA_ENV`:  `integration`, `stage`, or `production`
+* `CHANNEL` lets you specify the Cincinnati channel for version selection. Valid options include `nightly`, `candidate`, `fast`, and `stable`. By default, this is set to `candidate`. It is best practice to have several pipelines.  One that tests as far left as you can (e.g. nightlies) and one that tests candidate, fast or stable.  The idea behind this is that your left-most test pipeline will give you early warning of things that may break in the future, giving you time to react to failed test notifications and fix things.
+* `CONFIG`: Select environment, cloud provider and test suite. We have
+  * 3 test environments: integration (int), staging (stage), and production (prod).
+  * 3 providers: `rosa`, `gcp`, `aro`.  Each environment and each provider requires a separate prow job configuration.
+  * Test suite for your addon tests should be `addon-suite`
+    
+    The `CONFIGS` variable loads the config files defined in [osde2e](https://github.com/openshift/osde2e/tree/main/configs). The *test harness example* runs on `rosa` `stage` environment and executes `addon-suite`. If you want your job to run in a different environment, such as `int` or `prod`, or a different cloud provider, such as `gcp` or `aro`, you need to
+    * (A) change the job `name` key to include the proper environment and provider (i.e. `osde2e-<provider>-<environement>-<addon_name>-addon`) *and*
+    * (B) redefine the `CONFIGS` environment variable by replacing `rosa` and `stage` with the name of the appropriate provider and environment for your prow job.
+
+
+### More on Secrets
+
+For AWS or rosa clusters, you'll need to provide some additional details about your AWS account in a secret. In particular, you'll need to provide these values in your credentials secret:
+
+```
+aws-account
+aws-access-key
+aws-secret-access-key
+```
```

Comment:
Could we move this up higher. All addon testing should be on CCS accounts and the CICD team will only pass in the OCM Token.